go/vt/vtctl/reparentutil/emergency_reparenter.go

```diff
@@ -86,319 +108,194 @@ func NewEmergencyReparenter(ts *topo.Server, tmc tmclient.TabletManagerClient, l
 
 // ReparentShard performs the EmergencyReparentShard operation on the given
 // keyspace and shard.
-func (erp *EmergencyReparenter) ReparentShard(ctx context.Context, keyspace string, shard string, opts EmergencyReparentOptions) (*events.Reparent, error) {
-	opts.lockAction = erp.getLockAction(opts.NewPrimaryAlias)
-
+func (erp *EmergencyReparenter) ReparentShard(ctx context.Context, keyspace, shard string, opts EmergencyReparentOptions) (*events.Reparent, error) {
+	// First step is to lock the shard for the given operation
+	opts.lockAction = getLockAction(opts.newPrimaryAlias)
```

Comment:
see my other comment about keeping things as instance methods for clearer namespacing and avoiding namespace pollution

Reply:
Fixed